src/bvh/optimization.rs

```diff
@@ -0,0 +1,359 @@
+use bvh::*;
+use bounding_hierarchy::BHShape;
+use aabb::AABB;
+use std::collections::HashSet;
+
+// TODO Consider: Instead of getting the scene's shapes passed, let leaf nodes store an AABB
+// that is updated from the outside, perhaps by passing not only the indices of the changed
+// shapes, but also their new AABBs into optimize().
+
+impl BVH {
+    /// Optimizes the `BVH` by batch-reorganizing updated nodes.
+    /// Based on https://github.com/jeske/SimpleScene/blob/master/SimpleScene/Util/ssBVH/ssBVH.cs
+    ///
+    /// Needs all the scene's shapes, plus the indices of the shapes that were updated.
+    ///
+    pub fn optimize<Shape: BHShape>(&mut self,
+                                    refit_shape_indices: &HashSet<usize>,
+                                    shapes: &[Shape]) {
```

Comment:
Renamed type parameter from `T` to `Shape` in 3d29be6.
I don't see any `Vec<T: BHShape>` anywhere in `bvh.rs` though.